### DISCUSSION TITLE
Multiple docker stacks or single one?

### DISCUSSION THREAD

**Discussion by DennisGaida** [htpcBeginner/docker-traefik#212] [Ideas]
Currently everything is in one docker stack per device (https://github.com/htpcBeginner/docker-traefik/blob/master/docker-compose-t2.yml). I personally think that it is a bit unreadable having everything in one file.

Wouldn't it be advisable to have multiple docker stacks that are interconnected with common networks? Something that is already achieved with the `profiles` that are commented out. There could be one `core` docker stack that sets up Traefik & socket proxy and multiple other stacks that "make sense" such as a VPN stack that sets up everything related to that VPN.

What do you think? Keep everything in one file or make it more manageable with multiple files? Compete deployment would be easy since you can specify multiple compose files with `-f`. Another option would be to make use of `extends` (https://docs.docker.com/compose/extends/)

**Comment by Nickous** [comment]
I love the `profile` attribute.
I was using multiple docker-compose like @CVJoint
but now with profile...I prefer ONE full docker-compose file
